Sources/Build/Command.compile(ClangModule).swift

```diff
@@ -66,42 +47,80 @@ private extension ClangModule {
     }
 }
 
-private extension Sources {
-    func compilePathsForBuildDir(_ wd: String) -> [(filename: String, source: String, object: String, deps: String)] {
-        return relativePaths.map { source in
-            let path = Path.join(root, source)
-            let object = Path.join(wd, "\(source).o")
-            let deps = Path.join(wd, "\(source).d")
+/// A helper struct for ClangModule to compute basic
+/// flags needed to compile and link C language targets.
+struct ClangModuleBuildMeta {
```

Comment:
I wanted to implement something similar to how SwiftcTool works but since ClangTool compiles one file at a time I thought it wouldn't be right to have these methods there. The major problem this is solving (than method on ClangModule) is to avoid passing `prefix` everytime.